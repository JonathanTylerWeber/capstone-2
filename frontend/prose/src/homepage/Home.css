h1,
h2,
h4,
span,
p {
  color: white;
}

.head-p3 {
  font-size: 3em;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 1.5em;
}

.pen-container {
  width: 5em;
  height: 5em;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  margin-left: 50%;
  transform: translateX(-50%);
}

.pen {
  color: white;
  font-size: 3.5em;
}

.pen-container::before {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: #5bc19d;
  z-index: -1;
  border-radius: 20px;
}

.pen-container:hover .pen {
  color: black;
}

.pen-container:hover::before {
  background-color: white;
}


.img-col {
  display: flex;
  justify-content: center;
  align-items: center;
}

.img-col>div {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 262px;
  height: 262px;
  background-color: #5bc19d;
  border-radius: 40px;
  position: relative;
}

.fa-wand-magic-sparkles,
.fa-autoprefixer,
.fa-comment-dots,
.fa-face-smile-wink {
  height: 200px;
  width: 200px;
  color: white;
}

.wand,
.prefix,
.comment,
.smile {
  position: relative;
}

.wand::before,
.comment::before,
.smile::before,
.prefix::before {
  content: '';
  position: absolute;
  width: 262px;
  height: 262px;
  background-color: #5bc19d;
  z-index: -1;
  border-radius: 40px;
}


.title {
  font-size: 5em;
  font-weight: bold;
}

p,
span {
  font-size: 1.5em;
}

.head-p {
  font-size: 2em;
}

.head-p2 {
  font-size: 2em;
  margin-top: 1em;
}



h2 {
  font-size: 3em;
  font-weight: bold;
}

.head {
  margin-top: 10em;
  margin-bottom: 5em;
  text-align: center;
}

.text {
  margin-top: 10em;
  margin-bottom: 10em;
}

.foot {
  margin-top: 10em;
  margin-bottom: 15em;
  text-align: center;
}

.img-section {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.img-section>div {
  margin-top: 10em;
  margin-bottom: 10em;
}

.nav-link.sign {
  display: flex;
  justify-content: center;
  align-items: center;
}

.sign {
  margin-top: 3em;
}

.btn {
  background: #5bc19d;
  border: none;
  border-radius: 20px;
}

.signup-btn {
  font-size: 2em;
  border-radius: 40px;
  width: 6em;
  font-weight: bold;
}

.btn:hover {
  background: white;
  color: black;
}

button.navbar-toggler.collapsed {
  background: white;
  border: none;
}

button.navbar-toggler {
  background: white;
  border: none;
}

button.navbar-toggler:hover {
  background: #5bc19d;
}