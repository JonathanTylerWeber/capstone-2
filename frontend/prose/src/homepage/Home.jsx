import React from "react";
import "./Home.css"

function Home({ currentUser }) {
  return (
    <section className="col-md-8">
      <h1 className="font-weight-bold">
        Prose Perfector
      </h1>
      {currentUser ? (
        <>
          <p>Perfect your prose</p>
          <p>Welcome back, {currentUser.username}!</p>
        </>
      ) : (
        <p>We have jobs waiting for you. Please log in or sign up to explore.</p>
      )}
    </section>
  );
}

export default Home;