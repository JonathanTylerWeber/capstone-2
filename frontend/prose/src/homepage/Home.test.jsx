import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from './Home';

describe('Home Component', () => {
  it("matches snapshot", function () {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders title correctly', () => {
    render(<Home />);
    const titleElement = screen.getByText(/Perfect your prose/i);
    expect(titleElement).toBeInTheDocument();
  });

  test('renders description text correctly when user is logged in', () => {
    const currentUser = { username: 'TestUser' };
    render(<Home currentUser={currentUser} />);
    const welcomeText = screen.getByText(/Welcome back, TestUser!/i);
    expect(welcomeText).toBeInTheDocument();
  });

  test('renders description text correctly when user is not logged in', () => {
    render(<Home />);
    const descriptionText = screen.getByText(/Input your writing and the mood you want to convey/i);
    expect(descriptionText).toBeInTheDocument();
  });
});
