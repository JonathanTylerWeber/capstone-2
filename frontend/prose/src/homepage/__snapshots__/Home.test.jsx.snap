// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="home-section"
  >
    <div
      class="fluid container"
    >
      <div
        class="head"
      >
        <h1
          class="title"
        >
          Perfect your prose
        </h1>
        <p
          class="head-p2"
        >
          Input your writing and the mood you want to convey, 
          <br />
          and get a rating with tips to improve it.
        </p>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-1"
        />
        <div
          class="col-lg-5"
        >
          <div
            class="text"
          >
            <h2>
              Refine any type of writing with ease
            </h2>
            <p>
              Whether you're looking to express gratitude with a heartfelt thank-you letter or showcase your product with captivating descriptions, we've got you covered.
            </p>
          </div>
        </div>
        <div
          class="img-col col-lg-6"
        >
          <div
            class="prefix"
            style="opacity: 0;"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-autoprefixer "
              data-icon="autoprefixer"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M318.4 16l-161 480h77.5l25.4-81.4h119.5L405 496h77.5L318.4 16zm-40.3 341.9l41.2-130.4h1.5l40.9 130.4h-83.6zM640 405l-10-31.4L462.1 358l19.4 56.5L640 405zm-462.1-47L10 373.7 0 405l158.5 9.4 19.4-56.4z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-1"
        />
        <div
          class="col-lg-5"
        >
          <div
            class="text"
          >
            <h2>
              Customize the Tone
            </h2>
            <p>
              Select an adjective that best represents the tone you desire. Whether it's "gracious," "enticing," "persuasive," or any other adjective, we'll ensure your content reflects your desired style.
            </p>
          </div>
        </div>
        <div
          class="img-col col-lg-6"
        >
          <div
            class="smile"
            style="opacity: 0;"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-face-smile-wink "
              data-icon="face-smile-wink"
              data-prefix="far"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm177.6 62.1C192.8 334.5 218.8 352 256 352s63.2-17.5 78.4-33.9c9-9.7 24.2-10.4 33.9-1.4s10.4 24.2 1.4 33.9c-22 23.8-60 49.4-113.6 49.4s-91.7-25.5-113.6-49.4c-9-9.7-8.4-24.9 1.4-33.9s24.9-8.4 33.9 1.4zM144.4 208a32 32 0 1 1 64 0 32 32 0 1 1 -64 0zm165.8 21.7c-7.6 8.1-20.2 8.5-28.3 .9s-8.5-20.2-.9-28.3c14.5-15.5 35.2-22.3 54.6-22.3s40.1 6.8 54.6 22.3c7.6 8.1 7.1 20.7-.9 28.3s-20.7 7.1-28.3-.9c-5.5-5.8-14.8-9.7-25.4-9.7s-19.9 3.8-25.4 9.7z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-1"
        />
        <div
          class="col-lg-5"
        >
          <div
            class="text"
          >
            <h2>
              Get Feedback
            </h2>
            <p>
              Powered by the OpenAI API, we evaluate your text based on grammar and the chosen adjective, offering invaluable insights to help you refine and enhance your message.
            </p>
          </div>
        </div>
        <div
          class="img-col col-lg-6"
        >
          <div
            class="comment"
            style="opacity: 0;"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-comment-dots "
              data-icon="comment-dots"
              data-prefix="far"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M168.2 384.9c-15-5.4-31.7-3.1-44.6 6.4c-8.2 6-22.3 14.8-39.4 22.7c5.6-14.7 9.9-31.3 11.3-49.4c1-12.9-3.3-25.7-11.8-35.5C60.4 302.8 48 272 48 240c0-79.5 83.3-160 208-160s208 80.5 208 160s-83.3 160-208 160c-31.6 0-61.3-5.5-87.8-15.1zM26.3 423.8c-1.6 2.7-3.3 5.4-5.1 8.1l-.3 .5c-1.6 2.3-3.2 4.6-4.8 6.9c-3.5 4.7-7.3 9.3-11.3 13.5c-4.6 4.6-5.9 11.4-3.4 17.4c2.5 6 8.3 9.9 14.8 9.9c5.1 0 10.2-.3 15.3-.8l.7-.1c4.4-.5 8.8-1.1 13.2-1.9c.8-.1 1.6-.3 2.4-.5c17.8-3.5 34.9-9.5 50.1-16.1c22.9-10 42.4-21.9 54.3-30.6c31.8 11.5 67 17.9 104.1 17.9c141.4 0 256-93.1 256-208S397.4 32 256 32S0 125.1 0 240c0 45.1 17.7 86.8 47.7 120.9c-1.9 24.5-11.4 46.3-21.4 62.9zM144 272a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm144-32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zm80 32a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-1"
        />
        <div
          class="col-lg-5"
        >
          <div
            class="text"
          >
            <h2>
              Why Choose Us?
            </h2>
            <p>
              <b>
                User-Friendly Interface:
              </b>
               With a simple and intuitive interface, creating impactful content has never been easier.
            </p>
            <p>
              <b>
                Reliable Feedback:
              </b>
               Receive actionable feedback on your writing to help you improve and excel.
            </p>
            <p>
              <b>
                Powered by OpenAI:
              </b>
               Leveraging cutting-edge technology, we harness the power of the OpenAI API to deliver exceptional results.
            </p>
          </div>
        </div>
        <div
          class="img-col col-lg-6"
        >
          <div
            class="wand"
            style="opacity: 0;"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-wand-magic-sparkles "
              data-icon="wand-magic-sparkles"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M234.7 42.7L197 56.8c-3 1.1-5 4-5 7.2s2 6.1 5 7.2l37.7 14.1L248.8 123c1.1 3 4 5 7.2 5s6.1-2 7.2-5l14.1-37.7L315 71.2c3-1.1 5-4 5-7.2s-2-6.1-5-7.2L277.3 42.7 263.2 5c-1.1-3-4-5-7.2-5s-6.1 2-7.2 5L234.7 42.7zM46.1 395.4c-18.7 18.7-18.7 49.1 0 67.9l34.6 34.6c18.7 18.7 49.1 18.7 67.9 0L529.9 116.5c18.7-18.7 18.7-49.1 0-67.9L495.3 14.1c-18.7-18.7-49.1-18.7-67.9 0L46.1 395.4zM484.6 82.6l-105 105-23.3-23.3 105-105 23.3 23.3zM7.5 117.2C3 118.9 0 123.2 0 128s3 9.1 7.5 10.8L64 160l21.2 56.5c1.7 4.5 6 7.5 10.8 7.5s9.1-3 10.8-7.5L128 160l56.5-21.2c4.5-1.7 7.5-6 7.5-10.8s-3-9.1-7.5-10.8L128 96 106.8 39.5C105.1 35 100.8 32 96 32s-9.1 3-10.8 7.5L64 96 7.5 117.2zm352 256c-4.5 1.7-7.5 6-7.5 10.8s3 9.1 7.5 10.8L416 416l21.2 56.5c1.7 4.5 6 7.5 10.8 7.5s9.1-3 10.8-7.5L480 416l56.5-21.2c4.5-1.7 7.5-6 7.5-10.8s-3-9.1-7.5-10.8L480 352l-21.2-56.5c-1.7-4.5-6-7.5-10.8-7.5s-9.1 3-10.8 7.5L416 352l-56.5 21.2z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-1"
        />
        <div
          class="col-lg-10"
        >
          <div
            class="foot"
          >
            <h2>
              Start Creating Today:
            </h2>
            <span>
              Ready to elevate your content? Sign up now to unleash the full potential of your writing with Prose Perfector. It's free and always will be.
            </span>
            <a
              class="nav-link sign nav-link"
              exact="true"
              href="/signup"
            >
              <button
                class="btn signup-btn"
              >
                Sign up
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
